### PR TITLE
Exclude blacklistBranches in AvailabilityLabel + FindOnShelfModal

### PR DESCRIPTION
**Pull request by kasperbirch1**
#### This PR Exclude blacklistBranches in  AvailabilityLabel + FindOnShelfModal
Because blacklisted branches must not be included in the AvailabilityLabel or FindOnShelfModal accessibility check

#### Checklist

- [ ] My complies with [our coding guidelines](../docs/code_guidelines.md).
- [ ] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [ ] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

### CODE REVIEW

File: src/components/reservation/UserListItems.tsx [hunk 1]
```diff
@@ -47,8 +48,19 @@ const UserListItems: FC<UserListItemsProps> = ({
...
+    stringToArray: true
+  });
+
+  let pickupModalBranches = branches;
+  if (Array.isArray(blacklistBranches)) {
+    pickupModalBranches = excludeBlacklistedBranches(
+      branches,
+      blacklistBranches
+    );
+  }
```
Comment (spaceo) [inline src/components/reservation/UserListItems.tsx:62]: Could this go into a function in the helper called: `getPickupModalBranches()`?

File: src/components/find-on-shelf/FindOnShelfModal.tsx [hunk 2]
```diff
@@ -41,13 +42,18 @@ const FindOnShelfModal: FC<FindOnShelfModalProps> = ({
...
+  });
   const t = useText();
   const pidArray = getManifestationsPids(manifestations);
   const faustIdArray = pidArray.map((manifestationPid) =>
     convertPostIdToFaustId(manifestationPid)
   );
   const { data, isError, isLoading } = useGetHoldingsV3({
-    recordid: faustIdArray
+    recordid: faustIdArray,
+    ...(Array.isArray(blacklistBranches) ? { exclude: blacklistBranches } : {})
```
Comment (spaceo) [inline src/components/find-on-shelf/FindOnShelfModal.tsx:56]: simple and concise 👍

**Review by spaceo — COMMENTED** [review]
Looks good!
Have one sugesstion...

File: src/apps/material/material.dev.tsx [hunk 3]
```diff
@@ -26,6 +26,16 @@ export default {
       defaultValue: "1",
       control: { type: "text" }
     },
+    blacklistedBranchesConfig: {
+      name: "Blacklisted branches",
+      defaultValue: "FBS-751032,FBS-751031,FBS-751009,FBS-751027,FBS-751024",
+      control: { type: "text" }
+    },
```
Comment (kasperg) [inline src/apps/material/material.dev.tsx:33]: Please add seperate configuration for pickup and availability branches.

There are actually 3 different configurations for the different parts of the system that needs to be blacklisted.

**Review by kasperg — CHANGES_REQUESTED** [review]

File: src/components/availability-label/availability-label.tsx [hunk 4]
```diff
@@ -13,79 +14,90 @@ export interface AvailabilityLabelProps {
...
-  const { data, isLoading, isError } = useGetAvailabilityV3({
-    recordid: faustIds
-  });
+    const t = useText();
+    const { data, isLoading, isError } = useGetAvailabilityV3({
+      recordid: faustIds,
+      ...(Array.isArray(blacklistBranches)
+        ? { exclude: blacklistBranches }
+        : {})
+    });
```
Comment (kasperg) [inline src/components/availability-label/availability-label.tsx:37]: With the latest changes to `useConfig()`, `blacklistBranches` should always return an array of strings so either there is no need for this `Array.isArray()` check or the check should be different - e.g. checking for the length if we only want to pass the `exclude` parameter if there are any blacklisted branches configured.

File: src/components/availability-label/availability-label.tsx [hunk 5]
```diff
@@ -13,79 +14,90 @@ export interface AvailabilityLabelProps {
   handleSelectManifestation?: () => void | undefined;
   cursorPointer?: boolean;
 }
+export const AvailabilityLabel: React.FC<AvailabilityLabelProps> = withConfig(
```
Comment (kasperg) [inline src/components/availability-label/availability-label.tsx:17]: 🤔 I am still no sure whether this is the right place to add this. Have you tried updating the story to `component: withConfig(AvailabilityLabel)` instead?

If we cannot figure this out then perhaps we could leave this as is, merge it and talk to Mikkel about it on monday.

File: src/components/material/periodical/MaterialPeriodical.tsx [hunk 6]
```diff
@@ -18,8 +19,13 @@ const MaterialPeriodical: FC<MaterialPeriodicalProps> = ({
...
   selectPeriodicalHandler
 }) => {
+  const config = useConfig();
+  const blacklistBranches = config("blacklistedPickupBranchesConfig", {
+    transformer: "stringToArray"
+  });
   const { data, isLoading, isError } = useGetHoldingsV3({
-    recordid: [String(faustId)]
+    recordid: [String(faustId)],
+    ...(Array.isArray(blacklistBranches) ? { exclude: blacklistBranches } : {})
```
Comment (kasperg) [inline src/components/material/periodical/MaterialPeriodical.tsx:28]: Note the superfluous `isArray()` check here.

**Review by kasperg — CHANGES_REQUESTED** [review]
There are a few unnecessary array checks and then we have the continuous `withConfig()` saga for the tests.

Otherwise LGTM.

**Review by sinejespersen — APPROVED** [review]

**Pull request merged**